Feature/navigation integration

lawrenceDegoma opened the pull request:
# Complete Mobile App Foundation with Screen Implementation and Navigation Integration

##  Overview
This PR  implements the complete foundation of the app including all core screens, navigation system, and UI components. Building on the previous navigation cleanup and bundle optimization work, this PR follows the same conventions to avoid hardcoded styling values.

---

## Major Features Implemented

### Complete Screen Architecture
- **MapScreen**: Interactive campus map with parking lot selection and real-time occupancy (now without a legend nor lot labels)
- **ShortTermForecastScreen**: Hourly parking predictions with incident reporting
- **LongTermForecastScreen**: Extended parking forecasts and event planning
- **ProfileScreen**: User management with preferences and parking history
- **SettingsScreen**: App configuration and notification preferences (we don't really need this one anymore but I kept it around for reference)

### Navigation System
- **Header Component**: Unified header with conditional back navigation
- **Nested Navigation**: Stack navigator within tab navigator for seamless flow (Accessing short term forecast page when pressing on lot bubbles)
- **Navigation Buttons**: Contextual navigation across all screens
- **Deep Linking Ready**: Structured navigation types for future deep linking

### UI/UX Implementation
- **Modal System**: Filter and report modals with rich interactions
- **Chart Components**: Interactive hourly occupancy charts with confidence intervals
- **Theme Integration**: Cohesive color theming and consistent spacing and colors
- **Responsive Design**: Device-aware layouts with safe area handling (still working on this)

### Data Management & Visualization (Currently fabricated)
- **Mock Data Layer**: Structured parking lots, events, and forecast data
- **Real-time Updates**: Dynamic occupancy indicators and color-coded status
- **Interactive Charts**: Touch-friendly occupancy visualization
- **Event Integration**: Campus events affecting parking availability

---

## Screen-by-Screen Breakdown

### MapScreen
```typescript
// Key Features:
- Interactive CSULB campus map with parking lot overlays
- Real-time occupancy indicators with color-coded status
- Filter system for parking lot types and availability
- Navigation integration for turn-by-turn directions
- Responsive touch interactions for lot selection
```

**Core Functionality:**
- Campus map visualization with transparent unlabeled overlay
- Parking lot selection with real-time occupancy data
- Advanced filtering modal for lot types and availability
- Navigation button for GPS directions
- Color-coded occupancy indicators (Green: Available, Yellow: Limited, Red: Full)

### ShortTermForecastScreen
```typescript
// Key Features:
- Hourly parking predictions for next 24 hours
- Interactive charts with confidence intervals
- Campus event notifications affecting parking
- Incident reporting system with modal interface
- Accuracy indicators for prediction reliability
```

**Core Functionality:**
- Detailed hourly forecasting with accuracy indicators
- Interactive chart visualization with touch responsiveness
- Event notification system for campus activities
- Incident reporting modal with structured data collection
- Confidence intervals showing prediction reliability

### LongTermForecastScreen
```typescript
// Key Features:
- Extended 7-day parking forecasts
- Weekly pattern analysis and recommendations
- Event calendar integration
- Peak hours identification
- Comparative occupancy analysis
```

**Core Functionality:**
- Weekly parking pattern analysis and predictions
- Special event impact assessment
- Peak hour identification and recommendations
- Historical data comparison and trends
- Long-term parking planning assistance

### ProfileScreen
```typescript
// Key Features:
- User account management and preferences
- Parking history and usage analytics
- Favorite parking lots management
- Notification preferences configuration
- Account settings and privacy controls
```

**Core Functionality:**
- Comprehensive user profile management
- Parking usage history and analytics
- Favorite lots management with quick access
- Notification and alert preferences
- Privacy settings and data management

---

## 🔧 Technical Implementation Details

### Architecture Decisions
- **Component-Based Design**: Modular, reusable components for scalability
- **TypeScript Integration**: Full type safety across navigation and data structures
- **Performance Optimization**: Efficient rendering and state management
- **Accessibility Ready**: Structured for screen reader and accessibility support


### Design System Implementation
```typescript
// Updated Theme Constants:
export const SPACING = {
  xs: 4,     // Extra small spacing
  sm: 8,     // Small spacing
  md: 16,    // Medium spacing
  lg: 24,    // Large spacing
  xl: 32,    // Extra large spacing
  xxl: 40,   // 2x Extra large spacing
  xxxl: 48,  // 3x Extra large spacing
};

export const COLORS = {
  // Core brand colors
  primary: '#1E3A8A',
  secondary: '#F59E0B',
  
  // Status colors
  success: '#10B981',
  warning: '#F59E0B',
  error: '#EF4444',
  
  // UI colors
  background: '#F8FAFC',
  surface: '#FFFFFF',
  textPrimary: '#1F2937',
  // ... extended color palette
};
```

## Future Development Ready

### Next Phase Prep
- **API Integration**: Structured for backend connection
- **Push Notifications**: Ready for notification implementation
- **Analytics**: Event tracking infrastructure prepared
- **Offline Support**: Architecture supports offline capabilities

### Scalability Features
- **Component Library**: Reusable across future features
- **Type Safety**: Prevents runtime errors in development
- **Maintainability**: Clear code organization and documentation

---

DottL committed:
Resized + added logo v4

DottL committed:
Added folder structure for forecast screens

DottL committed:
Added mock data + UI interface for Long-Term Forecast

DottL committed:
Added banner + upcoming events

DottL committed:
Added calendar card w/ event + occupancy indicators

DottL committed:
Edited header

DottL committed:
Created reusable HourlyChart component

DottL committed:
Added basic components for short-term forecast screen

DottL committed:
refactor: extracted SubHeader component

DottL committed:
feat: create reusable SectionCard component

DottL committed:
feat: add SettingScreen w/ Notification and Appearance Card

lawrenceDegoma committed:
feat: update app icon and logo

- Add SharkParkV4.png for in-app header logo
- Add SharkParkV4AppIcon.png as app icon source
- Generate iOS app icons for all required sizes
- Generate Android app icons for all densities
- Update HomeScreen to use SharkParkV4.png
- Configure iOS Contents.json for custom icons

lawrenceDegoma committed:
Merge frontend changes and updated logo/app icon

- Merged latest frontend development work (settings screen, forecast screens, components)
- Integrated updated SharkParkV4 logo and app icons
- Resolved conflicts and combined team contributions

lawrenceDegoma committed:
Merge branch 'feature/bottom-navigation' into feature/navigation-integration

Merging frontend work with my navbar so we can start connecting screens

lawrenceDegoma committed:
in app navigation built working Long Term Forecasts, map view, and profile screens

- Remove unused screens: SearchScreen, LongTermForecastScreen_original
- Remove unused components: Button, Card
- Remove unused navigation: AppNavigation, ProfileStackNavigator
- Clean up assets: remove placeholder files
- Update navigation to use LongTermForecastScreen as HomeScreen
- Change home tab to 'Long Term' with forecast icon
- Set Map as default landing screen
- Remove Development Team ID from iOS project

lawrenceDegoma committed:
Merge branch 'main' into feature/navigation-integration

- Integrate theme constants from main branch
- Resolve navigation type conflicts
- Update ProfileScreen to use theme constants
- Keep existing navigation structure with Long Term and Map tabs
- Bundle optimization and font cleanup from main branch

lawrenceDegoma committed:
Complete theme integration with optimized constants and navigation setup

- Used attributes in theme.ts to template sizing and color schemes
- Set up navigation with Long Term Forecast, Map, and Profile screens
- Updated App.tsx with proper navigation structure
- Applied theme constants to Header, SubHeader, HourlyChart, SectionCard
- Integrated theme constants in all screens (Map, Profile, LongTerm, ShortTerm)
- Added .DS_Store to root .gitignore

lawrenceDegoma committed:
feat: Add modal components and integrate filtering functionality

- Copy FilterModal and ReportModal components from frontend-development branch
- Update both modals to use consistent theme.ts color constants
- Integrate ReportModal with ShortTermForecastScreen via FloatingActionButton
- Integrate FilterModal with MapScreen filter button functionality
- Add lot filtering logic to MapScreen for selective parking lot display
- Maintain TypeScript type safety across all modal integrations
- Ensure consistent design system alignment with existing app theme

lawrenceDegoma committed:
Added a transparent csulb app

lawrenceDegoma committed:
feat: implement navigation integration with Header component and navigation buttons

- Add conditional back navigation to Header component
- Integrate Header component across MapScreen and ShortTermForecastScreen
- Add NavigateButton component to both screens for navigation functionality
- Update theme constants with SPACING values for consistent positioning
- Enhance navigation types to support nested stack navigation
- Position navigation buttons symmetrically at bottom of screens
- Add campus map image asset for enhanced user experience
- Maintain existing filter and report button functionality
- Fix linting issues with unused variables

chuckmilton commented:
## Issues:
1. **No tests added:**
Because this is a 2,600+ line PR, we should include tests for our code.

## Nits:
1. **Hardcoded Values in Modals:**
Make sure to use theme constants to replace the hardcoded values in modals:
```typescript
// FilterModal.tsx
padding: 16,
borderRadius: 16,
maxWidth: 448,
paddingHorizontal: 24,
fontSize: 16,
```

2. **Console.log statements left in production code. Make sure to remove in the future:**
```typescript
// MapScreen.tsx
console.log('Navigate button pressed');
console.log('Applied filter with lots:', filteredLots);

// ProfileScreen.tsx
console.log('Logout pressed - functionality to be implemented');
```

3. **Large Image Assets:**
There are several large images that are included in the PR. In the future we should address this by compressing these images, using WebP formats, and/or remove unused images.

4. **Unused `scrollViewRef` in MapScreen:**
```typescript
const scrollViewRef = useRef<ScrollView>(null);
// ... ref={scrollViewRef} is set but never used
```

5. **Missing accessibility props:**
These elements should include accessibility labels:
```typescript
<TouchableOpacity onPress={() => onPress(lot)} activeOpacity={0.7}>
  <Text style={styles.lotText}>{lot.name}</Text>
</TouchableOpacity>
```
Should have something like:
```typescript
<TouchableOpacity 
  onPress={() => onPress(lot)} 
  activeOpacity={0.7}
  accessibilityLabel={`Parking lot ${lot.name}, ${lot.occupancy}% full`}
  accessibilityRole="button"
>
```

lawrenceDegoma committed:
refactor: cleanup and optimization

- Remove unused SettingScreen.tsx component
- Clean up console.log statements across all components
- Remove unused scrollViewRef from MapScreen
- Add comprehensive accessibility labels to FilterModal TouchableOpacity components
- Switch from PNG to WebP image format for better compression:
  - Replace CSULB campus map images with WebP versions
  - Replace SharkPark logo images with WebP versions
  - Remove old PNG files to reduce bundle size
- Improve code quality and production readiness
